### PR TITLE
[FIZZ-135] Map assignment-submissions

### PR DESCRIPTION
**Pull request by lalrg**

### CODE REVIEW

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [inline src/canvas-extractor/canvas_extractor/__main__.py:151]: I would have expected the base output directory to come from configuration, rather than being a hard-coded constant.

Would it be possible to more closely follow the pattern of either Google Classroom or Schoology in determining the file system? In fact, the LmsFilesystemProvider from Schoology could be extracted into a shared module so that all extractors can use the same code.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [inline src/canvas-extractor/canvas_extractor/__main__.py:51]: Is there any way to further differentiate the names `write_csv`, `write_multi_csv`, and `write_multi_tuple_csv` so that, when used below, it will be more obvious to the reader why one function was selected over the others?

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [inline src/canvas-extractor/canvas_extractor/__main__.py:156]: Would it make sense to move any of this code to a facade class?

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (bradbanister) [reply src/canvas-extractor/canvas_extractor/__main__.py:51]: A good idea.  Luis and I talked about using the Google Classroom code for writing data out, so that's where the naming comes from.  When we extract this out, we should improve the naming.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (bradbanister) [reply src/canvas-extractor/canvas_extractor/__main__.py:156]: Now that these have multiplied in the main file, I'd suggest moving these functions to their own modules in a similar pattern to the "api" and "mapping" module groupings.  Maybe under an "extract" grouping?

**Review by bradbanister — CHANGES_REQUESTED** [review]
overall looks good.  waiting on approval for discussion regarding minor restructure/rename.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:156]: I like the idea of a facade

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:150]: ✔️

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:157]: it is currently using GC approach, I have just added the .env variable support... The reason why we choose the GC over Schoology is because of GC is more generic and easier to maintain since it does not add a new method each time that we need to introduce a new path, but we can discuss about it.
@bradbanister, what do you think about this point?

Comment (bradbanister) [reply src/canvas-extractor/canvas_extractor/__main__.py:157]: I think what's missing is the parse-args stuff in the other extractors.  I think it's in a separate story?

Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:157]: That's part of FIZZ-56... the difference between the extractors regarding the filesystem, is that GC has a generic method and Schoology has a dedicated method for each file

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (lalrg) [reply src/canvas-extractor/canvas_extractor/__main__.py:51]: Is it okay to leave it for when we extract that logic in a shared library? or it should be done in this story?

Comment (bradbanister) [reply src/canvas-extractor/canvas_extractor/__main__.py:51]: I think it's ok to wait.  We should do the extract in the upcoming sprint.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [reply src/canvas-extractor/canvas_extractor/__main__.py:157]: Missing parse-arg stuff makes sense now. I'm not in a good spot right now for judging between the two (GC/Schoology) approaches, although my reaction-without-looking-at-code is that dedicated is better than generic, as it provides transparency to end-users and future programmers.

File: src/canvas-extractor/canvas_extractor/__main__.py
Comment (stephenfuqua) [reply src/canvas-extractor/canvas_extractor/__main__.py:51]: Please make sure that you write that as an acceptance criteria in an upcoming (or new) ticket.

**Pull request merged**
